lib/storage.ts

```diff
@@ -0,0 +1,144 @@
+/*
+Copyright 2016 Resin.io
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/**
+ * @module storage
+ */
+
+import * as Promise from 'bluebird';
+import getLocalStorage = require('./local-storage');
+
+/**
+ * @summary Get an instance of storage module
+ * @function
+ * @static
+ * @public
+ *
+ * @param {Object?} options - options
+ * @param {string?} options.dataDirectory - the directory to use for storage in Node.js. Ignored in the browser.
+ *
+ * @return {storage}
+ * @example
+ * const storage = require('resin-settings-storage')({
+ * 	dataDirectory: '/opt/cache/resin'
+ * })
+ */
+const getStorage = ({ dataDirectory }: { dataDirectory?: string } = {}) => {
+	const localStorage = getLocalStorage(dataDirectory);
+
+	/**
+	 * @summary Set a value
+	 * @function
+	 * @public
+	 *
+	 * @param {String} name - name
+	 * @param {*} value - value
+	 *
+	 * @return {Promise}
+	 *
+	 * @example
+	 * storage.set('token', '1234')
+	 */
+	const set = (name: string, value: any) =>
+		Promise.try(() => {
+			if (typeof value !== 'string') {
+				value = JSON.stringify(value);
+			}
+			return localStorage.setItem(name, value);
+		});
+
+	/**
+	 * @summary Get a value
+	 * @function
+	 * @public
+	 *
+	 * @param {String} name - name
+	 *
+	 * @return {Promise<*>} value or undefined
+	 *
+	 * @example
+	 * storage.get('token').then((token) => {
+	 * 	console.log(token)
+	 * });
+	 */
+	const get = (name: string) =>
+		Promise.try(() => {
+			// Run `node-localstorage` constructor to update
+			// internal cache of saved files.
+			// Without this, external changes to the data
+			// directory (with `fs` for example) will not
+			// be detected by `node-localstorage`.
+			if (typeof localStorage._init === 'function') {
+				localStorage._init();
+			}
+
+			let result: string | number | object | null = localStorage.getItem(name);
```

Comment:
Ah, I see, you've avoided that because the variable gets reused for parsed results below. I guess that's ok, I'd personally just immediately return rather than using the intermediate variable, but it doesn't really matter.